bin/create_juno_qc_report.py

```diff
@@ -0,0 +1,290 @@
+#         get_range(2.7, 3.23, row['Total length']), get_range(37.6, 38.2, row['GC (%)']), too_low(30000, row['N50']),
+#         "", too_low(30, row['Average coverage']), too_low(96, row['completeness']), exceed_max(4, row['contamination'])]
+
+#     elif genus == 'campylobacter':
+#         return ["", "", too_low(30, row['phred']), too_low(150, row['avg_sequence_length']), exceed_max(100, row['# contigs']), 
+#         get_range(1.5, 1.9, row['Total length']), get_range(29.5, 31.5, row['GC (%)']), too_low(30000, row['N50']),
+#         "", too_low(30, row['Average coverage']), too_low(96, row['completeness']), exceed_max(4, row['contamination'])]
+
+#     elif genus == 'yersinia':
+#         return ["", "", too_low(30, row['phred']), too_low(150, row['avg_sequence_length']), exceed_max(250, row['# contigs']), 
+#         get_range(3.9, 5.2, row['Total length']), get_range(46.2, 48.8, row['GC (%)']), too_low(30000, row['N50']),
+#         "", too_low(30, row['Average coverage']), too_low(96, row['completeness']), exceed_max(4, row['contamination'])]
+    
+#     return ["", "", "", "", "", "", "", "", "", "", "", ""]
+
+
+# def highlight_dataframe(dataframe: pd.DataFrame):
+#     """
+#     Apply conditional formatting to dataframe
+#     """
+#     styler = dataframe.style.apply(bacteria_colour, axis=1)
+        
+#     return styler
+
+def bp_mbp(dataframe: pd.DataFrame, col: str) -> pd.DataFrame:
+
+    """
+    Convert base pairs to mega base pairs
+    """
+    dataframe[col] = dataframe[col]/1000000
+    return dataframe
+
+def compile_report(species, phred, seq_len, quast, bbtools, checkm) -> pd.DataFrame:
+
+    """
+    Creates dataframe of required QC parameters from metadata, multiqc, quast, checkm and bbtools reports 
+    """
+
+    #gather dataframes
+    df_meta = get_metadata(species)
+    df_phred = get_phred_score(phred)
+    df_seq_length = get_sequence_len(seq_len)
+    df_quast = get_transposed_report(quast)
+    df_bbtools = get_bbtools_report(bbtools)
+    df_checkm = get_checkm_report(checkm)
+
+    #make list of dataframes
+    dfs = [df_meta, df_phred, df_seq_length, df_quast, df_bbtools, df_checkm]
+    # dfs = [df_meta, df_seq_length, df_quast, df_bbtools, df_checkm]
+    
+
+    #join dataframes
+    final_df = reduce(lambda  left,right: pd.merge(left,right,on=['sample'], how='outer'), dfs)
+
+    #apply function to convert base pairs to mega base pairs
+    final_df = bp_mbp(final_df, 'Total length')
+    #final_df2 = highlight_dataframe(final_df)
+    
+    return final_df
+
+def get_csv_report(dataframe):
+
+    """
+    Create csv format report from dataframe
+    """
+
+    dataframe.to_csv('juno_out.csv', index=False)
+
+def get_excel_report(dataframe):
+
+    """
+    Creates an excel format report with conditional formatting
+    """
+
+    writer = pd.ExcelWriter(sys.argv[-1], engine='openpyxl')
+
+    for genus in dataframe['genus'].unique():
```

Comment:
This is a kind of trivial function that is not used. Maybe it can be removed, or will it be useful in the future?

Reply:
Do you mean the create csv function?

Reply:
Yes, it highlighted the wrong lines

Reply:
I think we can remove it and add it back if someone ever needs a csv